osutil/sys_linux_test.go

```diff
@@ -70,3 +72,137 @@ func (s *sysSuite) TestSymlinkatAndReadlinkat(c *C) {
 	c.Assert(n, Equals, len("target"))
 	c.Assert(buf[:n], DeepEquals, []byte{'t', 'a', 'r', 'g', 'e', 't'})
 }
+
+func (s *sysSuite) TestExchangeFilesBothFilesMissing(c *C) {
+	// Create and open a temporary directory.
+	d := c.MkDir()
+	fd, err := syscall.Open(d, syscall.O_DIRECTORY, 0)
+	c.Assert(err, IsNil)
+	defer syscall.Close(fd)
+
+	missingA := filepath.Join(d, "missing-a")
+	missingB := filepath.Join(d, "missing-b")
+
+	// Both missing-a and missing-b do not exist.
+	c.Assert(missingA, testutil.FileAbsent)
+	c.Assert(missingB, testutil.FileAbsent)
+
+	// Exchange fails with ENOENT
+	err = osutil.ExchangeFiles(fd, "missing-a", fd, "missing-b")
```

Comment:
(nitpick) this could be "filepath.Base(missingA)" instead of hardcoding "missing-a" (and missing-b).